Update Docs for GA-Release

tomswartz07 opened the pull request:
This PR contains the doc updates related to the latest ga-release code.
Once this PR is approved, the process to merge this code upstream to the parent Pivotal repository will begin.

## Added:
- Updated template structure allowing for easy configuration of each page
- Utilization of `*.erb` variables, allowing single `config.yml` change to propagate to all affected templates
- Additional documentation for extra variables and configuration options
- Resolved issue with section headings not being displayed in respective page's TOC

## Removed:
- Removed redundant sections which are now better suited to template use with `partial`.

### Notes:
Local testing/build of the Bookbinder app requires the override of several (now-outdated) ruby gems. I did not commit the updated `Gemfile` and `Gemfile.lock` files because I was uncertain if these specific versions are necessary on the upstream repository. 
If testing/building locally, you can simply move/remove the `Gemfile.lock` files and update the Gemfile with the more recent and appropriate versions on your system.

## Testing
- Follow instructions documented in [the Readme](https://github.com/CrunchyData/docs-crunchy-data#how-to-use-bookbinder-to-view-your-docs) to serve docs locally
- Review resulting page on `localhost:4567` for any factual, typographical, or grammatical errors.

jessemcfarland commented on docs-content/_delta_restore.erb:
```diff
@@ -15,17 +15,17 @@ For guidance on using SSH to connect to servers in the service, see [Advanced Tr
     <pre class="terminal">
     sudo su - vcap
-    curl -sSL http<span>:</span>//localhost:8500/v1/catalog/service/postgresql-zone1?tag=primary \
+    curl -sSL http<span>:</span>//localhost:8500/v1/catalog/service/postgresql-0?tag=primary \
```
This can be reverted.

jessemcfarland commented on docs-content/_delta_restore.erb:
```diff
@@ -15,17 +15,17 @@ For guidance on using SSH to connect to servers in the service, see [Advanced Tr
 1. SSH into the `postgresql` server (identified above) using the `bosh` tool:
     <pre class="terminal">
-    bosh ssh postgresql-zone1 <index identified above>
+    bosh ssh postgresql 0<index identified above>
```
Don't need the `0` because it's referenced by `<index identified above>`

jessemcfarland commented on docs-content/_delta_restore.erb:
```diff
@@ -15,17 +15,17 @@ For guidance on using SSH to connect to servers in the service, see [Advanced Tr
     If using the Bosh V2 Client:
     <pre class="terminal">
-    bosh -e $ENV ssh postgresql-zone1 <index identified above>
+    bosh -e $ENV ssh postgresql 0 <index identified above>
```
Don't need the `0` because it's referenced by `<index identified above>`

jessemcfarland commented on docs-content/_pitr.erb:
```diff
@@ -15,17 +15,17 @@ For guidance on using SSH to connect to servers in the service, see the [Advance
     <pre class="terminal">
     sudo su - vcap
-    curl -sSL http<span>:</span>//localhost:8500/v1/catalog/service/postgresql-zone1?tag=primary \
+    curl -sSL http<span>:</span>//localhost:8500/v1/catalog/service/postgresql-0?tag=primary \
```
This can be reverted.

jessemcfarland commented on docs-content/_pitr.erb:
```diff
@@ -15,17 +15,17 @@ For guidance on using SSH to connect to servers in the service, see the [Advance
 1. SSH into the `postgresql` server (identified above) using the `bosh` tool:
     <pre class="terminal">
-    bosh ssh postgresql-zone1 <index identified above>
+    bosh ssh postgresql 0 <index identified above>
```
Don't need the `0` because it's referenced by `<index identified above>`.

jessemcfarland commented on docs-content/_pitr.erb:
```diff
@@ -15,17 +15,17 @@ For guidance on using SSH to connect to servers in the service, see the [Advance
     If using the Bosh V2 Client:
     <pre class="terminal">
-    bosh -e $ENV ssh postgresql-zone1 <index identified above>
+    bosh -e $ENV ssh postgresql 0 <index identified above>
```
Don't need the `0` because it's referenced by `<index identified above>`.

jessemcfarland commented on docs-content/installing.html.md.erb:
```diff
@@ -71,7 +71,7 @@ If unsure, use the defaults.
 If unsure, use the defaults.
 
-1. Choose an AZ to which deploy these services.
+1. Choose an AZ in which deploy these services.
```
Should be: Choose an AZ in which **to** deploy these services.

wilybrace commented:
### Introduction
* Version is 03.090510.107, not 1.90510.107
### Feature Overview
* We updated the plans to better reflect their names as presented, but the tables in **Suggested Plan Configuration** and **Default Cluster Configurations** have the old names.
* Does it make sense to have the Tools section if those tools are not available directly through the tile?
### Installation and Configuration
* Numbering scheme breaks in configuration sections where graphic is after bullet 1 (OL shows 1. 1. 2. 3. ...)
* **Configure Plan Properties** table should reflect the plan naming convention in the feature overview
* Should we not direct people looking at PivNet partner documentation to PivNet for downloading of stemcells? I'm not sure if the version lag reflects compatibility testing on Pivotal's part, but it seems more consistent than directing to bosh.io
### Using Crunchy PostgreSQL
* According to the `spec` file, postgis defaults to false
* We seem to be missing the entirety of the new `pgc_` optional parameters.
* Something is broken in the CLI examples in **Creating a New Service** - it appears to be interpreting the underscores as markdown formatting.
* Since `APP_NAME`, `SERVICE_INSTANCE_NAME`, and `SERVICE_NAME` are placeholders, they should be enclosed in angle brackets to be consistent with CLI help conventions, not converted into environment variables.
* SERVICE_INSTANCE_NAME and SERVICE_NAME seem to refer to the same value. If so, they should be consistent. While longer, SERVICE_INSTANCE_NAME seems more correct.

tomswartz07 commented:
>Version is 03.090510.107, not 1.90510.107

Updated in next commit.

>We updated the plans to better reflect their names as presented, but the tables in Suggested Plan Configuration and Default Cluster Configurations have the old names.

I can update it as such, but (unless I'm mistaken) this table is directly referencing the installation procedure's steps, where the 'plan' sizes and properties must be defined during setup.

i.e.: 
![](http://crunchy-docs-pivotal.pde.east.paas.geointservices.io/crunchy/images/small-plan-properties.png)

It might be useful to convert this table to be a partial, however, since it's used and referenced in several locations. Doing so will allow for easier updating of these values and plan names. 

>Does it make sense to have the Tools section if those tools are not available directly through the tile?

They're not available as part of the tile installation, but it's useful to note that we provide them. Basically, it points out; "ask for these tools if you want them"

> Numbering scheme breaks in configuration sections where graphic is after bullet 1 (OL shows 1. 1. 2. 3. ...)

Fixed in next commit.

>Configure Plan Properties table should reflect the plan naming convention in the feature overview

Will be updated as part of the partial template use. Still same issue as raised above: names are matching the setup procedure steps.

>Should we not direct people looking at PivNet partner documentation to PivNet for downloading of stemcells? I'm not sure if the version lag reflects compatibility testing on Pivotal's part, but it seems more consistent than directing to bosh.io

This is a direct link to the stemcell downloads, we can certainly include links to what they are and what they do, but I'd argue that having a link to the page where the downloads are hosted is most useful in the context of the installation documentation procedure to prevent unnecessary delay.


>According to the spec file, postgis defaults to false

Will update accordingly. 
I believe that we have previously overridden it in the ODB template for the service-adapter:
`crunchy-postgresql-service-adapter.properties.service_adapter.postgis: true`
Can you confirm that it's set to false in the service-adapter as well?

>We seem to be missing the entirety of the new pgc_ optional parameters.

Looks like a missing partial. Will be updated.

>Something is broken in the CLI examples in Creating a New Service - it appears to be interpreting the underscores as markdown formatting.

Appears to be an issue with the CSS/layout stuff for the 'terminal' display. Will convert to standard code fencing blocks. 

>Since APP_NAME, SERVICE_INSTANCE_NAME, and SERVICE_NAME are placeholders, they should be enclosed in angle brackets to be consistent with CLI help conventions, not converted into environment variables.

Fixed in next commit.

>`SERVICE_INSTANCE_NAME` and `SERVICE_NAME` seem to refer to the same value. If so, they should be consistent. While longer, `SERVICE_INSTANCE_NAME` seems more correct.

Fixed in next commit.

>`jdbc_read_uri` seems to suffer from Markdown sickness as well (Integrating Applications > Service Credentials)

Will be fixed in next commit as well

wilybrace commented:
* The tables are still showing Xlarge instead of extra-large from the plan names or Extra-Large to match the case of the side menu. 
The "Click on <size> Plan Properties" made sense to exactly match the graphic and should not have been altered, the request was for the tables to match the plan names.

* If we're going to keep the tools section, we should add cf-grafana to the list and ensure the roadmap has a separate release artifact for cf-pgloader covered.

* Remove `pgc_max_connections` and `pgc_port`

### Using Crunchy PostgreSQL...
* In **Creating a New Service** - it looks like a line break went missing between the `cf bind service` and the `cf restage` - Similarly the block of commands for bullet 1 of this section was more readable with its line breaks between individual parameters. (ref: http://docs.pivotal.io/partners/crunchy/using.html)
* **Service Credentials** has the following additional keys when the `monitoring` parameter was used during provisioning:
    * `prometheus_host`: host of the Prometheus server for monitoring data
    * `prometheus_port`: connection port for the Prometheus server
    * `prometheus_username`
    * `prometheus_password`

Pedantic notes:
* Introduction shows the version as 3.090510.107, the release notes show the correct, official version number as 03.090510.107
* "Extra Large" Plan Properties is referred to as "Extra-Large" everywhere else.

kyle-nevins commented:
We also need to update the release date to 01/24/2018. The deployment won't be on PivNet until tomorrow.
